spec/services/anno_auth_validator_spec.rb

```diff
@@ -0,0 +1,43 @@
+require 'rails_helper'
+require 'webmock/rspec'
+
+RSpec.describe AnnoAuthValidator do
+
+  context 'when authorize is called' do
+    describe 'validate canvas and user' do
+      before(:each) do
+        Rails.application.config.JWT_Password = 'abc123'
+        Rails.application.config.JWT_Canvas_Verification_Url = 'http://testserver.com'
+      end
+
+      it 'check valid JWT' do
+        JWT.stub(:decode).and_return([{"group_id" => "10", "user_id" => "100"}])
```

Comment:
I'm not sure what version of RSpec we're using but the latest version stubs methods using the code:

```ruby
allow(obj).to receive(:method).and_return(stuff)
```

Just FYI.